src/core/public/notifications/toasts/toasts_api.tsx

```diff
@@ -143,6 +141,15 @@ export class ToastsApi implements IToasts {
    * @returns a {@link Toast}
    */
   public add(toastOrTitle: ToastInput) {
+    if (typeof toastOrTitle !== 'string') {
+      const list = this.toasts$.getValue();
+      const existingToast = list.find((toast) => toast.id === toastOrTitle.id);
```

Comment:
nit, I think this would be more readable if it were something like
```suggestion
    if (typeof toastOrTitle !== 'string') {
      const toastObject = toastOrTitle;
      const list = this.toasts$.getValue();
      const existingToast = list.find((toast) => toast.id === toastObject.id);
```

to indicate that the whole point of the conditional is to type-narrow from `toastOrTitle` to `toast`